### PR TITLE
feat:middleware timeline

### PR DESCRIPTION
**Pull request by thapasya-m**
## PR Checklist
Please check that your PR fulfills the following requirements:

- [x] The commit message follows [our guidelines](https://github.com/pesto-students/batch-10-tripen/wiki/Commit-message-guidelines)
- [ ] Tests for the changes have been added (for bug fixes / features)
- [ ] Docs have been added / updated (for bug fixes / features)
- [ ] GIFs/Videos/Images added (compulsory for frontend changes)

## PR Type
What kind of change does this PR introduce? 

- [ ] Bugfix
- [x] Feature
- [ ] Tests changed / added / updated
- [ ] Code style update (formatting, local variables)
- [ ] Refactoring (no functional changes, no api changes)
- [ ] Build related changes
- [ ] CI related changes
- [ ] Documentation content changes
- [ ] Infrastructure changes
- [ ] Dependency upgrade
- [ ] Other... Please describe:

## Issue Number: 
#

## Does this PR introduce a breaking change?
- [ ] Yes
- [x] No

## Other information

### CODE REVIEW

File: server/.babelrc [hunk 1]
```diff
@@ -1,4 +1,7 @@
 {
-  "presets": ["env"],
```
Comment (p-racoon) [inline server/.babelrc]: What happened to presets?

File: server/src/middlewares/timeline.js [hunk 2]
```diff
@@ -0,0 +1,30 @@
+import mongoose from 'mongoose';
+import Timeline from '../models/timeline';
+
+export const verifyTimelineId = (req, res, next) => {
+  const { id } = req.params;
+  try {
+    if(mongoose.Types.ObjectId.isValid(id)) {
+      next();
+    } else {
+      return res.status(500).json({ msg: `Given param Id ${id} is invalid` });
```
Comment (karansapolia) [inline server/src/middlewares/timeline.js]: won't 400 (Bad Request) be a better status message here?

File: server/src/middlewares/timeline.js [hunk 3]
```diff
@@ -0,0 +1,30 @@
...
+  }
+}
+export const verifyTimelineObject = (req, res, next) => {
+  try {
+    const timelineObj = req.body;
+    const mdl = new Timeline(timelineObj);
+    const error = mdl.validateSync();
+    if(typeof error !== undefined) {
+      console.log('error', error.message);
+      res.status(500).json({ msg: `Given req body is invalid timeline object. ${error.message}`});
```
Comment (karansapolia) [inline server/src/middlewares/timeline.js]: again. If request is incorrect/invalid, we should throw a 400 right?

File: server/src/middlewares/timeline.test.js [hunk 4]
```diff
@@ -0,0 +1,66 @@
...
+  describe('verifyTimelineId: req handler create', () => {
+    let mw;
+    beforeAll(() => {
+      mw = verifyTimelineId;
+    });
+    it('should return a function', () => {
+      expect(mw.length).toBe(3);
+    });
+    it('should take 3 args', () => {
+      expect(mw).toBeInstanceOf(Function);
```
Comment (karansapolia) [inline server/src/middlewares/timeline.test.js]: checking for number of arguments or type of `mw` ?

File: server/src/middlewares/timeline.test.js [hunk 5]
```diff
@@ -0,0 +1,66 @@
...
+      expect(nextSpy).toHaveBeenCalledTimes(1);
+    });
+    describe('should not call next()',() => {
+      it('response status 500',async ()=> {
+        const req = mockReq('invalid_id');
+        const res = mockRes();
+        const nextSpy = jest.fn();
+        await verifyTimelineId(req, res, nextSpy);
+        expect(nextSpy).toHaveBeenCalledTimes(0);
+        expect(res.status).toHaveBeenCalledWith(500);
```
Comment (karansapolia) [inline server/src/middlewares/timeline.test.js]: change similarly to 400

File: server/src/middlewares/timeline.test.js [hunk 6]
```diff
@@ -0,0 +1,66 @@
+import { verifyTimelineId, verifyTimelineObject } from './timeline';
+
+describe('middleware: timeline', () => {
+  describe('verifyTimelineId: req handler create', () => {
+    let mw;
+    beforeAll(() => {
+      mw = verifyTimelineId;
+    });
+    it('should return a function', () => {
```
Comment (karansapolia) [inline server/src/middlewares/timeline.test.js]: You have switched the bodies of this and the test below it

File: server/src/middlewares/timeline.test.js [hunk 7]
```diff
@@ -0,0 +1,66 @@
...
+        });
+      });
+    });
+  });
+  describe('verifyTimelineObject: req handler create', () => {
+    let mw;
+    beforeAll(() => {
+      mw = verifyTimelineObject;
+    });
+    it('should return a function', () => {
```
Comment (karansapolia) [inline server/src/middlewares/timeline.test.js]: bodies of tests switched here too

**Comment by karansapolia** [comment]
Hey @thapasya-m, did you get some time to review the comments and resolve the conflicts with master? Hope you can take this up at the earliest :slightly_smiling_face:

File: server/src/middlewares/timeline.test.js (hunk 4 above)
Comment (thapasya-m) [reply server/src/middlewares/timeline.test.js]: there was, as you pointed of swapped test description. fixed in latest commit

File: server/src/middlewares/timeline.test.js (hunk 6 above)
Comment (thapasya-m) [reply server/src/middlewares/timeline.test.js]: changed in latest commit

File: server/src/middlewares/timeline.test.js (hunk 5 above)
Comment (thapasya-m) [reply server/src/middlewares/timeline.test.js]: done

File: server/src/middlewares/timeline.test.js (hunk 4 above)
Comment (thapasya-m) [reply server/src/middlewares/timeline.test.js]: type od middleware. swapped test description

**Pull request merged**